{
    "url": "https://api.github.com/repos/rapidsai/cudf/issues/15926",
    "repository_url": "https://api.github.com/repos/rapidsai/cudf",
    "labels_url": "https://api.github.com/repos/rapidsai/cudf/issues/15926/labels{/name}",
    "comments_url": "https://api.github.com/repos/rapidsai/cudf/issues/15926/comments",
    "events_url": "https://api.github.com/repos/rapidsai/cudf/issues/15926/events",
    "html_url": "https://github.com/rapidsai/cudf/issues/15926",
    "id": 2335305068,
    "node_id": "I_kwDOBWUGps6LMe1s",
    "number": 15926,
    "title": "[BUG] The `ndarray` wrapper in `cudf.pandas` doesn't have `__array_interface__`.",
    "user": {
        "login": "trivialfis",
        "id": 16746409,
        "node_id": "MDQ6VXNlcjE2NzQ2NDA5",
        "avatar_url": "https://avatars.githubusercontent.com/u/16746409?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/trivialfis",
        "html_url": "https://github.com/trivialfis",
        "followers_url": "https://api.github.com/users/trivialfis/followers",
        "following_url": "https://api.github.com/users/trivialfis/following{/other_user}",
        "gists_url": "https://api.github.com/users/trivialfis/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/trivialfis/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/trivialfis/subscriptions",
        "organizations_url": "https://api.github.com/users/trivialfis/orgs",
        "repos_url": "https://api.github.com/users/trivialfis/repos",
        "events_url": "https://api.github.com/users/trivialfis/events{/privacy}",
        "received_events_url": "https://api.github.com/users/trivialfis/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 599626559,
            "node_id": "MDU6TGFiZWw1OTk2MjY1NTk=",
            "url": "https://api.github.com/repos/rapidsai/cudf/labels/bug",
            "name": "bug",
            "color": "d73a4a",
            "default": true,
            "description": "Something isn't working"
        },
        {
            "id": 6815620706,
            "node_id": "LA_kwDOBWUGps8AAAABlj4eYg",
            "url": "https://api.github.com/repos/rapidsai/cudf/labels/cudf.pandas",
            "name": "cudf.pandas",
            "color": "984DFB",
            "default": false,
            "description": "Issues specific to cudf.pandas"
        }
    ],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 0,
    "created_at": "2024-06-05T08:59:04Z",
    "updated_at": "2024-06-05T16:41:37Z",
    "closed_at": null,
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "**Describe the bug**\r\nThe `ndarray` wrapper in `cudf.pandas` doesn't have `__array_interface__`, but has `__cuda_array_interface__` instead. I assume this is caused by it being backed by cupy. Discovered in https://github.com/rapidsai/cudf/issues/15906#issuecomment-2149220361 .\r\n\r\n**Steps/Code to reproduce bug**\r\n``` python\r\n# file: reproduce.py\r\nimport cudf\r\n\r\nprint(cudf.__version__)         # 24.08.00a101\r\n\r\nimport pandas as pd\r\ndf = pd.DataFrame({\"a\": [1, 2, 3], \"b\": [4, 5, 6]})\r\narr = df.values\r\n\r\nprint(arr.__cuda_array_interface__) # good\r\nprint(arr.__array_interface__)  # AttributeError\r\n```\r\nRun:\r\n``` sh\r\npython -m cudf.pandas ./reproduce.py \r\n```\r\n\r\n**Expected behavior**\r\nI don't know what to expect in this case. This might actually help XGBoost to work with cuDF pandas wrapper since we can directly fetch the CUDA memory, albeit in a hacky way exploiting an unexpected difference between `cudf.pandas` and `pandas`. It's still a bug instead of a defined behavior. I will leave it to cuDF devs to decide what to do with it.\r\n\r\n**Environment overview (please complete the following information)**\r\n - Environment location: Bare-meta\r\n - Method of cuDF install: conda\r\n\r\n**Environment details**\r\n- cuDF: 24.08.00a101",
    "closed_by": null,
    "reactions": {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/15926/reactions",
        "total_count": 0,
        "+1": 0,
        "-1": 0,
        "laugh": 0,
        "hooray": 0,
        "confused": 0,
        "heart": 0,
        "rocket": 0,
        "eyes": 0
    },
    "timeline_url": "https://api.github.com/repos/rapidsai/cudf/issues/15926/timeline",
    "performed_via_github_app": null,
    "state_reason": null
}